src/main/java/com/sumologic/log4j/SumoLogicAppender.java

```diff
@@ -234,6 +239,12 @@ public void append(LogEvent event)
 
     try {
       queue.add(message);
+      if (event.getLevel().isInRange(Level.FATAL, Level.ERROR)) {
+        if(!queue.hasError()) {
+          queue.setError(true);
+          this.flusher.interrupt();
```

Comment:
the goal is to provide a best effort to flush error events. you are right that it could miss some events between clearing the flag and add (which is fine as they will be picked up during next flush). The assumption here is that error events are relatively rare as compared to info events. 

i thought above using PriorityBlockingQueue, but as you mentioned, it has a lot of overhead.